src/org/labkey/test/tests/SampleTypeTest.java

```diff
@@ -123,12 +125,47 @@ private void doSetup()
     protected void doCleanup(boolean afterTest)
     {
         super.doCleanup(afterTest);
-
+        if (previousSampleStatusFlag != null)
+            SampleTypeHelper.setSampleStatusEnabled(previousSampleStatusFlag);
         // If you are debugging tests change this function to do nothing.
         // It can make re-running faster but you need to valid the integrity of the test data on your own.
 //        log("Do nothing.");
     }
 
+    @Test
+    public void testDeleteSampleTypeWithLockedSamples()
+    {
+        SampleTypeHelper sampleTypeHelper = new SampleTypeHelper(this);
+        previousSampleStatusFlag = SampleTypeHelper.setSampleStatusEnabled(true);
```

Comment:
Same comment as before, maybe put in a more general doSetup function.

Reply:
Hah!  Well, this one was supposed to go away since I moved it into the other class.